storage/fanout.go

```diff
@@ -512,8 +537,11 @@ func (c *genericMergeSeriesSet) Next() bool {
 		c.currentLabels = c.heap[0].At().Labels()
 		for len(c.heap) > 0 && labels.Equal(c.currentLabels, c.heap[0].At().Labels()) {
 			set := heap.Pop(&c.heap).(genericSeriesSet)
-			if c.querier != nil && c.querier.IsFailedSet(set) {
```

Comment:
@bwplotka To address your comment in here https://github.com/prometheus/prometheus/pull/7251#discussion_r430309497, querier was optional previously.